plugins/woocommerce/includes/wc-template-functions.php

```diff
@@ -4204,6 +4204,61 @@ function wc_set_hooked_blocks_version() {
 	add_option( $option_name, WC()->version );
 }
 
+/**
+ * Attach functions that listen to theme switches.
+ *
+ * @since 9.7.0
+ *
+ * @param string    $old_name Old theme name.
+ * @param \WP_Theme $old_theme Instance of the old theme.
+ * @return void
+ */
+function wc_after_switch_theme( $old_name, $old_theme ) {
+	wc_set_hooked_blocks_version_on_theme_switch( $old_name, $old_theme );
+	wc_update_store_notice_visible_on_theme_switch( $old_name, $old_theme );
+}
+
+/**
+ * Update the Store Notice visibility when switching themes:
+ * - When switching from a classic theme to a block theme, disable the Store Notice.
+ * - When switching from a block theme to a classic theme, re-enable the Store Notice
+ *   only if it was enabled last time there was a switchi from a classic theme to a block theme.
+ *
+ * @since 9.7.0
+ *
+ * @param string    $old_name Old theme name.
+ * @param \WP_Theme $old_theme Instance of the old theme.
+ * @return void
+ */
+function wc_update_store_notice_visible_on_theme_switch( $old_name, $old_theme ) {
+	$enable_store_notice_in_classic_theme_option = 'woocommerce_enable_store_notice_in_classic_theme';
+	$is_store_notice_active_option               = 'woocommerce_demo_store';
+
+	if ( ! $old_theme->is_block_theme() && wc_current_theme_is_fse_theme() ) {
+		/*
+		 * When switching from a classic theme to a block theme, check if the store notice is active,
+		 * if it is, disable it but set an option to re-enable it when switching back to a classic theme.
+		 */
+		$store_notice_is_active = get_option( $is_store_notice_active_option, 'no' );
+
+		if ( 'yes' === $store_notice_is_active ) {
+			update_option( $is_store_notice_active_option, 'no' );
+			add_option( $enable_store_notice_in_classic_theme_option, 'yes' );
```

Comment:
I know we are using `yes` and `no` here but in WC there are `wc_bool_to_string` and `wc_string_to_bool` functions [here](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/wc-formatting-functions.php#L18-L42). Shall we use these instead?

Reply:
Good point, better to be cautious just in case. :slightly_smiling_face: I fixed it here b6c795c38595d1aa8dc5ff30501b55047c48abe1. I also made it so we use the existing `is_store_notice_showing()` function instead of getting the option directly.